src/main/core/prim/misc.scala

```diff
@@ -0,0 +1,451 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.core
+package prim
+
+//scalastyle:off number.of.types
+case class _and() extends Reporter with Pure {
+  override def syntax =
+    Syntax.reporterSyntax(
+      left = Syntax.BooleanType,
+      right = List(Syntax.BooleanType),
+      ret = Syntax.BooleanType,
+      precedence = Syntax.NormalPrecedence - 6)
+}
+case class _any() extends Reporter {
+  override def syntax =
+    Syntax.reporterSyntax(
+      right = List(Syntax.AgentsetType),
+      ret = Syntax.BooleanType)
+}
+case class _ask() extends Command {
+  override def syntax =
+    Syntax.commandSyntax(
+      right = List(Syntax.AgentsetType | Syntax.AgentType, Syntax.CommandBlockType),
+      agentClassString = "OTPL",
```

Comment:
Ordinarily, yes. But when the parser dynamically instantiates primitives, it looks for a zero-argument constructor, which unfortunately scala does not generate when a default argument is supplied.